### PR TITLE
Extensions Endpoints Guides Security Improvements

### PR DESCRIPTION
**Pull request by phazonoverload**
Prior to this PR, we're showing people how to create unauthenticated API proxies, but the example we use is Twilio - an API that costs money to use. Whoops. 

This PR does the following: 
- Replaces the API for the 'public endpoint' to a public read-only API. 
- Adds a `req.accountability` check to the Twilio guide. 
- Makes it clear in the listing that there are three endpoint guides with varying security: public, authenticated, permission-based. 

@paescuj will go through all guides on Friday to make sure they are correct.

### CODE REVIEW

File: docs/guides/extensions/endpoints-api-proxy.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+		router.get('/*', (req, res) => {
+			axios.get(`https://pokeapi.co/api/v2/${req.url}`).then((response) => {
+				res.json(response.data);
+			}).catch((error) => {
+				res.send(error);
+			});
+		});
+	},
+};
```
Comment (br41nslug) [inline docs/guides/extensions/endpoints-api-proxy.md:138]:
```suggestion
import axios from 'axios';

export default {
	id: 'pokeapi',
	handler: (router, { env }) => {
		router.get('/*', (req, res) => {
			axios.get(`https://pokeapi.co/api/v2/${req.url}`).then((response) => {
				res.json(response.data);
			}).catch((error) => {
				res.send(error);
			});
		});
	},
};
```

**Review by br41nslug — CHANGES_REQUESTED** [review]
The original complete code would only work if you rename it to `index.cjs` which is not something we want to recommend i think. However importing `axios` like this also isn't great as it will try to bundle `axios` into the extension. Is it an option to use something else like `fetch` (requires node 18+) or `node-fetch` (for node<18)

**Review by paescuj — APPROVED** [review]

File: docs/guides/extensions/endpoints-api-proxy-twilio.md [hunk 2]
```diff
@@ -79,28 +71,105 @@ The standard way to create an API route is to specify the method and the path. R
...
-		res.send(error);
-	});
+router.get('/*', async (req, res) => {
+	try {
+		const response = await fetch(new URL(req.url, twilioHost), { headers });
+
+		if (response.ok) {
+			res.json(await response.json());
+		} else {
+			res.status(response.status).send(response.statusText);
```
Comment (br41nslug) [inline docs/guides/extensions/endpoints-api-proxy-twilio.md:81]:
```suggestion
			res.status(response.status);
			res.send(response.statusText);
```
I think this is more readable

File: docs/guides/extensions/hooks-validate-number-twilio.md [hunk 3]
```diff
@@ -29,10 +29,15 @@ npm install twilio
 
 ## Build the Hook
 
-Create a collection called Customers with a text field called `phone_number`. This hook will be used to validate the
-item when a record is saved.
+Create a collection called Customers with a text field called `phoneNumber`. This hook will be used to validate the item
```
Comment (br41nslug) [inline docs/guides/extensions/hooks-validate-number-twilio.md:32]:
```suggestion
Create a collection called Customers with a text field called `phone_number`. This hook will be used to validate the item
```

I dont think we should start using pascalCase for database fields in the docs.

Comment (paescuj) [reply docs/guides/extensions/hooks-validate-number-twilio.md:32]: Will revert 👍

**Review by br41nslug — APPROVED** [review]

**Review by paescuj — APPROVED** [review]

**Pull request merged**